Jenkinsfile

```diff
@@ -1,17 +1,41 @@
 // For ci.jenkins.io
 // https://github.com/jenkins-infra/documentation/blob/master/ci.adoc
 
+def imageName = 'jenkinsciinfra/ath'
+
+def container
+/* Assuming we're not inside of a pull request or multibranch pipeline */
+if (!(env.CHANGE_ID || env.BRANCH_NAME)) {
+    node('docker') {
+        stage('Prepare Container') {
+            timestamps {
+                checkout scm
+                sh 'git rev-parse HEAD > GIT_COMMIT'
+                shortCommit = readFile('GIT_COMMIT').take(6)
+                def imageTag = "${env.BUILD_ID}-build${shortCommit}"
+                echo "Creating the container ${imageName}:${imageTag}"
+                container = docker.build("${imageName}:${imageTag}", 'src/main/resources/ath-container')
+            }
+        }
+
+        stage('Publish container') {
+            infra.withDockerCredentials {
+                timestamps { container.push() }
+            }
+        }
+    }
+} else { // PR or multibranch pipeline
 
     for (int i = 0; i < (BUILD_NUMBER as int); i++) {
-      milestone()
+        milestone()
     }
     def splits = splitTests count(10)
     def branches = [:]
     for (int i = 0; i < splits.size(); i++) {
         int index = i;
         branches["split${i}"] = {
             stage("Run ATH - split${index}") {
-                node('docker && highmem'){
+                node('docker && highmem') {
                     checkout scm
                     def image = docker.build('jenkins/ath', 'src/main/resources/ath-container')
```

Comment:
In order to remove confusion, It would  be better to specify a tag different than what already exist on hub.docker.com.
And avoid the default 'latest'.